index.d.ts

```diff
@@ -84,7 +84,8 @@ declare module 'monk' {
     findOneAndUpdate<U = T>(
       query: TQuery,
       update: Object,
-      options?: Object
+      options?: Object,
+      callback?: (err: Error, data: U) => void
     ): Promise<U | undefined>
```

Comment:
I don't think it's returning a promise when there is a callback. So maybe we need an override definition

Reply:
i noticed i had a few formatting changes in there. i disabled everything except this projects settings via vscode however i noticed it changed a couple lines.  if you believe this is in error i can revert those tiny bits. otherwise i think it's good to go now.